### PR TITLE
Add support for Custom Roles

### LINKED ISSUE #81: Support for Custom Roles
Add support for [azurerm_role_definition](https://registry.terraform.io/providers/hashicorp/azurerm/latest/docs/resources/role_definition).

### PR DESCRIPTION
**Pull request by kimfy**
This PR  resolves #81 
- Implement Custom Roles

Requirements:
- [x] Creation: Create custom roles
- [x] Outputs: Mirror outputs from native resource [link](https://registry.terraform.io/providers/hashicorp/azurerm/latest/docs/resources/role_definition#attributes-reference)
- [x] Pass custom roles to Terraform Cloud Workspace as Terraform Variables (see tfe.tf)

### CODE REVIEW

**Comment by ghost** [comment]
Should we not add a test for this?

**Comment by kimfy** [comment]
> Should we not add a test for this?

Can you help write the test? See ./tests/role_definitions_test.tf for a starter.

**Review by ghost — APPROVED** [review]

**Comment by kimfy** [comment]
@Blinq-SanderBlom see updated PR description

**Comment by ghost** [comment]
@kimfy See the new commits. I also update the test to pass the role_definitions to the TFC wokspace

File: test/README.md [hunk 1]
```diff
@@ -1,22 +1,103 @@
 # Terraform Station Tests
 
-This folder contains all testing done with Station. As of now, these are ran locally from a developer's machine. This folder is supposed to make development and testing much easier. In the future this will run in GHA as well.
+## Overview
 
-## How to run
+This folder contains all tests for the Terraform Station module. Currently, tests are executed locally on a developer's machine, with plans to integrate them into GitHub Actions (GHA) in the future. The aim is to simplify the development and testing process for contributors.
```
Comment (kimfy) [inline test/README.md:5]: Is this still correct? @Blinq-SanderBlom

Comment (ghost) [reply test/README.md:5]: I'll update the readme in the other PR where I add the automated tests

**Pull request merged**